### DISCUSSION TITLE
Supported Ruby versions

### DISCUSSION THREAD

**Discussion by tenderlove** [tenderlove/tenderjit#98] [Q&A]
What versions of Ruby should we support?  I thought that Ruby 3.1.0 would not really change instructions much, but it seems there's been a bunch of activity, and now we have more instructions to support.  See https://github.com/tenderlove/tenderjit/pull/97

My gut says TenderJIT should only target Ruby master plus only the most recent release of Ruby.  Does anyone have any thoughts on this?  Or ways we can be more resilient to change?

**Comment by 64kramsystem** [comment]
> My gut says TenderJIT should only target Ruby master plus only the most recent release of Ruby. Does anyone have any thoughts on this? Or ways we can be more resilient to change?

I'm not familiar with the Ruby, so I can't really estimate what's the rate of internal change of Ruby versions, however, an option is to decided what to support along the way, based on the maintenance cost.

Once a minor version is supported (e.g. v3.0), if patch versions change little or nothing, and consequently cost little or no support, it would help people who are late to upgrade the Ruby version (in particular, because, AFAIK, YJIT won't be enabled by default at least until v3.2).

With this strategy, there's still a case for dropping a given old version, even if its implicit cost is low/nothing: when maintaining multiple versions requires undesirable design overhead (I reckon, complex conditionals).
